### PR TITLE
Change token management transactions parser according to the sdk-specs

### PR DESCRIPTION
**Pull request by popenta**

### CODE REVIEW

File: src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.ts [hunk 1]
```diff
@@ -71,221 +67,239 @@ export class TokenManagementTransactionsOutcomeParser {
...
-        const tokenIdentifier = this.extractTokenIdentifier(event);
-
-        const encodedRoles = event.topics.slice(3);
+        const events = this.findEventsByIdentifier(transactionOutcome, "ESDTSetRole");
 
-        let roles: string[] = [];
-        for (const role of encodedRoles) {
-            roles = roles.concat(this.decodeTopicAsString(role));
-        }
+        return events.map((event) => {
```
Comment (danielailie) [inline src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.ts:75]: can you extract a method here?

File: src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.ts [hunk 2]
```diff
@@ -71,221 +67,239 @@ export class TokenManagementTransactionsOutcomeParser {
...
+    }[] {
         this.ensureNoError(transactionOutcome.transactionLogs.events);
 
-        const event = this.findSingleEventByIdentifier(transactionOutcome, "ESDTNFTCreate");
-        const tokenIdentifier = this.extractTokenIdentifier(event);
-        const nonce = this.extractNonce(event);
-        const amount = this.extractAmount(event);
+        const events = this.findEventsByIdentifier(transactionOutcome, "ESDTNFTCreate");
+
+        return events.map((event) => {
```
Comment (danielailie) [inline src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.ts:94]: also here

File: src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.ts [hunk 3]
```diff
@@ -71,221 +67,239 @@ export class TokenManagementTransactionsOutcomeParser {
...
-        const burntSupply = this.extractAmount(event);
-
-        return {
-            userAddress: userAddress,
-            tokenIdentifier: tokenIdentifier,
-            nonce: nonce,
-            burntSupply: burntSupply,
-        };
+        const events = this.findEventsByIdentifier(transactionOutcome, "ESDTLocalBurn");
+
```
Comment (danielailie) [inline src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.ts:137]: also here

File: src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.ts [hunk 4]
```diff
@@ -6,57 +6,53 @@ import { bufferToBigInt } from "../smartcontracts/codec/utils";
...
+            );
         }
 
-        return { tokenIdentifier: tokenIdentifier, roles: roles };
+        return registerEvents.map((registerEvent, index) => {
+            const tokenIdentifier = this.extractTokenIdentifier(registerEvent);
+            const encodedRoles = setRoleEvents[index].topics.slice(3);
+            const roles = encodedRoles.map((role) => this.decodeTopicAsString(role));
+            return { tokenIdentifier, roles };
+        });
```
Comment (andreibancioiu) [inline src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.ts:55]: Could it be possible (in real-life) that the order of `registerEvents` is not correlated with the order of `setRoleEvents`?

Comment (popenta) [reply src/transactionsOutcomeParsers/tokenManagementTransactionsOutcomeParser.ts:55]: Hopefully not

**Review by andreibancioiu — APPROVED** [review]

**Review by danielailie — APPROVED** [review]

**Pull request merged**